### ISSUE TITLE
Child buildings not being placed at correct rotation

### ISSUE THREAD

**Issue by Talyda** [ldtteam/minecolonies#10260]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Are you using the latest MineColonies Version?

- [X] I am running the latest **beta/release** version of MineColonies for my Minecraft version.
I am also running the latest versions of other mods that are part of my problem.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [X] Single Player
- [ ] Multi Player

### Minecraft Version

1.21

### MineColonies Version

.707

### Structurize Version

.751

### Current Behavior

Shire style tavern, with built in restaurant, places restaurant 180 degrees off where it should be (and was, in 1.20).
Apparently Moobien had reported it to Ray for Aquanautica th builder and residence.
Saw it reported here https://github.com/ldtteam/minecolonies/issues/5006#issuecomment-2366865307 and tested myself cos I thought it might be the old "mirroring" problem, but I didn't mirror or rotate at all and had same thing happen.

### Expected Behavior

Restaurant places in intended direction.

### Reproduction Steps

Build shire tavern
order in built restaurant to be built

### Logs

https://gist.github.com/Talyda/2c3f8e096bf3163a86706585a7b68a35

### Footer

---
## Viewers
* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Mikec78660** [comment]
I have seen this with the barracks in the medieval oak style pack. 2 of the 4 towers were rotated incorrectly.

**Comment by Talyda** [comment]
> I have seen this with the barracks in the medieval oak style pack. 2 of the 4 towers were rotated incorrectly.

In 1.21 as well?

**Comment by Mikec78660** [comment]
Yes. Also I can't remember if I mirrored the barricks or not, but either way seems like the towers should have mirrored as well.

> > I have seen this with the barracks in the medieval oak style pack. 2 of the 4 towers were rotated incorrectly.
> 
> In 1.21 as well?

**Comment by yi-fan-song** [comment]
This is also the case with the pagoda style barracks, with 2/4 barrack towers in the back, they will be built oriented the wrong way.
tested in creative, no rotation nor flip.

minecolonies 1.1.719

Edit: I tested some more with different styles, it happens often with barracks because every barrack tower, when they use the same template, will have the same orientation.
